### PR TITLE
Updated ConvertPrecision pass not to make extra copy of Constant data

### PR DESCRIPTION
**Pull request by GlebKazantaev**

### CODE REVIEW

File: inference-engine/src/transformations/src/transformations/convert_precision.cpp [hunk 1]
```diff
@@ -300,17 +300,22 @@ std::shared_ptr<Node> change_constant_precision(std::shared_ptr<opset4::Constant
...
-                       if (val > std::numeric_limits<dst_type>::max()) {
-                           return std::numeric_limits<dst_type>::max();
-                       } else {
-                           return static_cast<dst_type>(val);
-                       }
-                   });
-    return std::make_shared<ngraph::opset4::Constant>(PREC_TO, constant->get_shape(), final_data);
+    for (size_t i = 0; i < size; ++i) {
+        const auto & val = src_data[i];
+        if (val > std::numeric_limits<dst_type>::max()) {
```
Comment (ilyachur) [inline inference-engine/src/transformations/src/transformations/convert_precision.cpp:312]: I guess we should also add: `if (val < std::numeric_limits<dst_type>::min()) {`

Comment (GlebKazantaev) [reply inference-engine/src/transformations/src/transformations/convert_precision.cpp:312]: If val for example has uint64 type and dst_type int32 then comparison will be wrong as for comparison will be used type with bigger size -> uint64. So then min int32 value-2^32 will be converted to unsigned and comparison will be incorrect.

Comment (ilyachur) [reply inference-engine/src/transformations/src/transformations/convert_precision.cpp:312]: But what if we want to convert int64 to int32?

**Review by ilyachur — APPROVED** [review]
LGTM, please create a ticket to check how this transformation works with cases when min of source precision is less than minimum of destination precision

**Pull request merged**